### ISSUE TITLE
pwntools is slow to import, because of update.py

### ISSUE THREAD

**Issue by kokjo** [Gallopsled/pwntools#736]
https://github.com/Gallopsled/pwntools/pull/659

```
$ time python -c "from pwn import *"
real    0m1.624s
user    0m1.484s
sys 0m0.120s
```

because of "import pip"

**Comment by TethysSvensson** [comment]
The problem is that we are importing datetime, json, pip and pkg_resources whenever we import pwn. That is **slow**.

I will create a PR to fix it.

**Comment by TethysSvensson** [comment]
I will also make it a part of my release process in the future, though I would need to have a good way of testing it.

@zachriggle: Is there any reliable way to get timing out of Travis, or is it too non-deterministic?

**Comment by TethysSvensson** [comment]
In general we seem to be importing a **bunch** of libraries that we will only need in very specific functions. This slows down pwntools by **a lot**. I will create a follow-up PR that fixes a lot of these (though we might have to discuss the individual ones on that PR).

**Comment by zachriggle** [comment]
We already have timing information on Travis.  There's no tests for it, since that'd lead to high failures (e.g. it may genuinely take 0.2s normally, but 1s under heavy load).

**Comment by TethysSvensson** [comment]
Fixed in #738. Release pending.
